src/main/resources/Development-documentation/gerrit-sceanrios.md

```diff
@@ -0,0 +1,224 @@
+- Pushes to `refs/for/[branch name]`
+- Receives comments from reviewer
+- Updates code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]`
+- Gets ok from the reviewer
+- Hits the submit button in Gerrit
+
+Events sent:
+
+- SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
+- SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
+- SCS(`E3`) at submit with `CHANGE` link set `E2`
+
+Commit history after submit:
+
+```git
+* C2 Review changes
+* C0 Commit from previous user
+```
+
+Trigger table:
+
+| Event | Triggered by | Comment |
+| ---   | ---          | ---     |
+| E0    |              | Submit  |
+| E1    | P1           |
+| E2    | P2           |
+| E3    |              | Submit  |
+
+SCC Event table:
+
+| Event | BASE | PREVIOUS_VERSION |
+| ---   | ---  | ---              |
+| E1    | E0   |
+| E2    | E0   | E1               |
+
+## Create patch set with rebase - CASE2
+
+Preconditions:
+
+- Previous review submits to  the master with hash `C0`
+- The plugin sends SCS with id `E0`
+
+The user does the following:
+
+- Creates a branch
+- Updates the code
+- Squashes the commits to one commits (`C1`)
+- Pushes to `refs/for/[branch name]` (`P1`)
+- Receives comments from reviewer
+- Updates code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]`  (`P2`)
+- Gets ok from the reviewer
+- Forced to rebase the code due to `C01` (`C3`)  event `E01`
+- Pushes to `refs/for/[branch name]`  (`P3`)
+- Hits the submit button in Gerrit
+
+Events sent:
+
+- SCC(`E1`) sent for `P1` push with `BASE` link set to `E0`
+- SCC(`E2`) sent for `P2` push with `PREVIOUS_VERSION` link set `E1` and `BASE` link set to `E0`
+- SCC(`E3`) sent for `P3` push with `PREVIOUS_VERSION` link set `E2` and `BASE` link set to `E01`
+- SCS(`E4`) sent at submit with `CHANGE` link set `E3`
+
+Commit history after submit:
+
+```git
+* C3 Rebased changes
+* C01 Other code change
+* C0 Commit from previous user
+```
+
+Trigger table:
+
+| Event | Triggered by | Comment |
+| ---   | ---          | ---     |
+| E0    |              | Submit  |
+| E1    | P1           |
+| E2    | P2           |
+| E3    | P3           |
+| E4    |              | Submit  |
+
+SCC Event table:
+
+| Event | BASE | PREVIOUS_VERSION |
+| ---   | ---  | ---              |
+| E1    | E0   |
+| E2    | E0   | E1               |
+| E3    | E01  | E2               |
+
+## Create patch set with merge - CASE3
+
+Preconditions:
+
+- Previous review submits to  the master with hash `C0`
+- The plugin sends SCS with id `E0`
+
+The user does the following:
+
+- Creates a branch
+- Updates the code
+- Squashes the commits to one commits (`C1`)
+- Pushes to `refs/for/[branch name]` (`P1`)
+- Receives comments from reviewer
+- Updates code
+- Does `commit --amend` (`C2`)
+- Pushes to `refs/for/[branch name]`  (`P2`)
+- Gets ok from the reviewer
+- Forced to do a merge due to `C01` (`C3`) event `E01`
```

Comment:
"Forced to do a merge due to `C01` (`C3`) event `E01`"

What happened in C01 and E01? This is first mention and states due to. Something happened before and I do not know what.

Reply:
Will the changes in 041b7cd help you understand it better?